lighthouse-core/gather/driver.js

```diff
@@ -890,11 +888,8 @@ class Driver {
     await this._beginNetworkStatusMonitoring(url);
     await this._clearIsolatedContextId();
 
-    // These can 'race' and that's OK.
-    // We don't want to wait for Page.navigate's resolution, as it can now
-    // happen _after_ onload: https://crbug.com/768961
-    this.sendCommand('Page.enable');
-    this.sendCommand('Emulation.setScriptExecutionDisabled', {value: disableJS});
+    await this.sendCommand('Page.enable');
+    await this.sendCommand('Emulation.setScriptExecutionDisabled', {value: disableJS});
     // No timeout needed for Page.navigate. See #6413.
     this._innerSendCommand('Page.navigate', {url});
```

Comment:
ok, after talking this through for a little while, we think we do want to incorporate the `Page.navigate` errors into the promise chain. It's a little ugly, so if anyone has any other structure ideas, please chime in :)

Basically it would be something like

```js
const waitforPageNavigateCmd = this._innerSendCommand('Page.navigate', {url});
// ...
let pageWaiter = Promise.resolve();
if (waitForNavigated) {
  pageWaiter = this._waitForFrameNavigated();
} else if (waitForLoad) {
  // ...
  pageWaiter = this._waitForFullyLoaded(/* ... */);
}

await Promise.all([
  waitforPageNavigateCmd,
  pageWaiter,
]);
```
- If `Page.navigate` has no problem, as before it doesn't block or interfere with observing the page load
- If `Page.navigate` throws, it wins the `Promise.all` with a rejection, which leads to lighthouse exit.

The main issue I see (besides reduced readability) is that we don't clean up the `_waitForFullyLoaded` machinery. This is a fatal error so it doesn't matter in most clients, but are we going to have to actually clean up to support when LH is run as a module?

Reply:
also during error testing we discovered that `_waitForFrameNavigated` should probably also have a timeout instead of blindly waiting on the event.

Reply:
> we discovered that _waitForFrameNavigated should probably also have a timeout

WDYT about having an option to `this.once` that does the timeout bit?

Reply:
> is that we don't clean up the _waitForFullyLoaded machinery

why don't we just await `waitForPageNavigateCmd` as the ultimate very last thing? the only thing we want to happen is get the errors into the chain, right? if we do it as the very last thing, then it can't prevent other cleanup

Reply:
> WDYT about having an option to this.once that does the timeout bit?

yeah, good idea. We have at least one or two places we will happily wait forever on an event.

Reply:
though the service worker one I'm thinking of only waits forever on `about:blank`, which adding an `_waitForFrameNavigated ` timeout will fix :)

Reply:
> why don't we just await `waitForPageNavigateCmd` as the ultimate very last thing? the only thing we want to happen is get the errors into the chain, right? if we do it as the very last thing, then it can't prevent other cleanup

This would definitely be easier to follow.

The ordering will be a little wonky (I'd imagine a `Page.navigate` failure would end up usually failing on another command within `_waitForFullyLoaded` or by hitting the 45 second timeout), but ordering is already an issue with `Promise.all()` (and without cancelling `_waitForFullyLoaded`), and we've already all agreed that `Page.navigate` rarely if ever fails :)